### PR TITLE
Review code before merging, when merged should code should be deployable

### PR DESCRIPTION
**Pull request by wensle**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+from fastapi import FastAPI
+from fastapi.responses import PlainTextResponse
+from starlette.middleware.cors import CORSMiddleware
+
+from controller.text_to_videoController import router as textToVido
```
Comment (wensle) [inline app/main.py]: The `textToVido` variable contains a typo.

Comment (wensle) [inline app/main.py]: Use the naming convention for Python, see https://stackoverflow.com/questions/159720/what-is-the-naming-convention-in-python-for-variable-and-function

File: app/models/tts.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
+import os
+import requests
+import azure.cognitiveservices.speech as speechsdk
+from services.utils import removeTags,split_into_sentences
```
Comment (wensle) [inline app/models/tts.py]: For `removeTags` use the naming convention for Python, see https://stackoverflow.com/questions/159720/what-is-the-naming-convention-in-python-for-variable-and-function

File: app/models/tts_open_source.py
Comment (wensle) [inline app/models/tts_open_source.py:1]: Delete this file

File: app/mouthCoordinates/mouthcoordinates.csv
Comment (wensle) [inline app/mouthCoordinates/mouthcoordinates.csv:1]: Why was this file changed? Did we change the coordinates? If so, we could say something about how to do this in the documentation.

File: app/services/gentelPhonemes.py
Comment (wensle) [inline app/services/gentelPhonemes.py:1]: File name contains typo

Comment (wensle) [inline app/services/gentelPhonemes.py:1]: For the filename use the naming convention for Python modules, see https://softwareengineering.stackexchange.com/a/308976

File: app/services/gentelPhonemes.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
+import requests
+import json
+from services.utils import removeTags
```
Comment (wensle) [inline app/services/gentelPhonemes.py]: For `removeTags` Use the naming convention for Python, see https://stackoverflow.com/questions/159720/what-is-the-naming-convention-in-python-for-variable-and-function

File: app/services/lazykhHumanImager.py
Comment (wensle) [inline app/services/lazykhHumanImager.py:1]: For the filename use the naming convention for Python modules, see https://softwareengineering.stackexchange.com/a/308976

File: app/services/lazykhHumanImager.py [hunk 4]
```diff
@@ -1,16 +1,11 @@
...
 import argparse
 import os.path
 import pygame
+import numpy as np
+import math
+import shutil
 import random
-from utils import getFilenameOfLine, getTopic, capitalize
-
+from utils import removeTags, getFilenameOfLine, getTopic, capitalize
```
Comment (wensle) [inline app/services/lazykhHumanImager.py]: For `getFilenameOfLine`, `getTopic`, and `capitalize` use the naming convention for Python, see https://stackoverflow.com/questions/159720/what-is-the-naming-convention-in-python-for-variable-and-function

File: app/services/lazykhMusicAdder.py
Comment (wensle) [inline app/services/lazykhMusicAdder.py:1]: For the filename use the naming convention for Python modules, see https://softwareengineering.stackexchange.com/a/308976

File: app/services/lazykhVideoDrawer.py
Comment (wensle) [inline app/services/lazykhVideoDrawer.py:1]: For the filename use the naming convention for Python modules, see https://softwareengineering.stackexchange.com/a/308976

File: app/services/lazykhVideoFinisher.py
Comment (wensle) [inline app/services/lazykhVideoFinisher.py:1]: For the filename use the naming convention for Python modules, see https://softwareengineering.stackexchange.com/a/308976

File: app/services/lazykhschduler.py
Comment (wensle) [inline app/services/lazykhschduler.py:1]: For the filename fix the typo and use the naming convention for Python modules, see https://softwareengineering.stackexchange.com/a/308976

File: app/services/lazykhschduler.py [hunk 5]
```diff
@@ -0,0 +1,159 @@
+import argparse
+import os.path
+import json
+import numpy as np
+import random
+
+def addPhoneme(p, t):
```
Comment (wensle) [inline app/services/lazykhschduler.py]: Use the naming convention for Python, see https://stackoverflow.com/questions/159720/what-is-the-naming-convention-in-python-for-variable-and-function

File: app/services/lazykhschduler.py [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+    global prevPhoneme
+    global f
+    if p != prevPhoneme:
+        strings[4] += (str.format('{0:.3f}', t)+",phoneme,"+p+"\n")
+    prevPhoneme = p
+
+def pickNewPose(t):
```
Comment (wensle) [inline app/services/lazykhschduler.py]: Use the naming convention for Python, see https://stackoverflow.com/questions/159720/what-is-the-naming-convention-in-python-for-variable-and-function

File: app/services/utils.py [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+import os
+import re
+
+def split_into_sentences(text):
+    """Splits the text into sentences based on some punctuation marks."""
+    sentences = re.split(r'[.!?]\s+', text)
+    return sentences
+
+
+def removeTags(script):
```
Comment (wensle) [inline app/services/utils.py]: Use the naming convention for Python, see https://stackoverflow.com/questions/159720/what-is-the-naming-convention-in-python-for-variable-and-function

File: app/services/utils.py [hunk 8]
```diff
@@ -0,0 +1,164 @@
...
+        newScript = newScript.replace("\n ","\n")
+    while " \n" in newScript:
+        newScript = newScript.replace(" \n","\n")
+    while newScript[0] == " ":
+        newScript = newScript[1:]
+
+    return newScript
+
+
+def creat_randome_name():
```
Comment (wensle) [inline app/services/utils.py]: Contains typo

File: app/services/utils.py [hunk 9]
```diff
@@ -0,0 +1,164 @@
...
+        frames=folder_path+prefix+'_frames'
+        # remove the directory and all its contents
+        shutil.rmtree(frames)
+    except PermissionError:
+        print(f"{prefix} is in use. Skipping deletion.")
+    except Exception as e:
+        print(f"Error deleting {prefix}: {e}")
+
+
+def getFilenameOfLine(line):
```
Comment (wensle) [inline app/services/utils.py]: Use the naming convention for Python, see https://stackoverflow.com/questions/159720/what-is-the-naming-convention-in-python-for-variable-and-function

File: app/services/utils.py [hunk 10]
```diff
@@ -0,0 +1,164 @@
...
+    topic = getTopic(line)
+    return re.sub(r'[^A-Za-z0-9 -]+', '',  topic.lower())
+
+def getTopic(stri):
```
Comment (wensle) [inline app/services/utils.py]: Use the naming convention for Python, see https://stackoverflow.com/questions/159720/what-is-the-naming-convention-in-python-for-variable-and-function

File: note [hunk 11]
```diff
@@ -0,0 +1,3 @@
+you will have to create this file named env_vars.txt in this directory and the content should be as follows to make sure docker orks:
```
Comment (wensle) [inline note]: Start the sentence with a Capital letter and fix typo

File: note
Comment (wensle) [inline note:1]: Why don't we use .env files and python-dotenv? I would use that instead. Rename this file to example.env (if we are going to use python-dotenv).

File: app/controller/text_to_videoController.py
Comment (wensle) [inline app/controller/text_to_videoController.py:1]: For the filename use the naming convention for Python modules, see https://softwareengineering.stackexchange.com/a/308976

File: app/conftest.py
Comment (wensle) [inline app/conftest.py:1]: Remove file

File: app/services/lazykhschduler.py [hunk 12]
```diff
@@ -0,0 +1,159 @@
...
+    global pose
+    global prevPose
```
Comment (wensle) [inline app/services/lazykhschduler.py]: Use the naming convention for Python, see https://stackoverflow.com/questions/159720/what-is-the-naming-convention-in-python-for-variable-and-function

File: app/services/lazykhschduler.py [hunk 13]
```diff
@@ -0,0 +1,159 @@
...
+    global POSE_COUNT
+    global prevPhoneme
```
Comment (wensle) [inline app/services/lazykhschduler.py]: Use the naming convention for Python, see https://stackoverflow.com/questions/159720/what-is-the-naming-convention-in-python-for-variable-and-function

File: app/services/lazykhschduler.py
Comment (wensle) [inline app/services/lazykhschduler.py:1]: I tried to mark them all individually, but that's not feasible. For every variable name (where it's applicable) Use the naming convention for Python, see https://stackoverflow.com/questions/159720/what-is-the-naming-convention-in-python-for-variable-and-function

File: app/services/utils.py [hunk 14]
```diff
@@ -0,0 +1,164 @@
...
+    # Read the video file as binary data
+        def iterfile():  # 
+             with open(path+'_final.mp4', mode="rb") as file_like:  # 
+                yield from file_like  # 
+
+        return StreamingResponse(iterfile(), media_type="video/mp4")
+
+
+
+def delete_temprory_files(folder_path, prefix):
```
Comment (wensle) [inline app/services/utils.py]: Fix typo

File: app/conftest.py
Comment (MohammedN-hash) [reply app/conftest.py:1]: This is for testing but since we have no testing we can delete it for now

File: app/controller/text_to_videoController.py
Comment (MohammedN-hash) [reply app/controller/text_to_videoController.py:1]: Renamed

File: app/main.py (hunk 1 above)
Comment (MohammedN-hash) [reply app/main.py]: Fixed the typo

Comment (MohammedN-hash) [reply app/main.py]: Adopted style:
ClassName, function_name, var_name

File: app/mouthCoordinates/mouthcoordinates.csv
Comment (MohammedN-hash) [reply app/mouthCoordinates/mouthcoordinates.csv:1]: only the name of the file was changed

File: app/services/gentelPhonemes.py
Comment (MohammedN-hash) [reply app/services/gentelPhonemes.py:1]: Fixed

File: app/services/lazykhHumanImager.py
Comment (MohammedN-hash) [reply app/services/lazykhHumanImager.py:1]: Same comment in the pull reques.

File: app/services/lazykhHumanImager.py (hunk 4 above)
Comment (MohammedN-hash) [reply app/services/lazykhHumanImager.py]: This is the original code from the lazyKH, I will change them later on when we have second verion working.

File: app/services/lazykhschduler.py
Comment (MohammedN-hash) [reply app/services/lazykhschduler.py:1]: Fixed

File: app/services/lazykhschduler.py (hunk 5 above)
Comment (MohammedN-hash) [reply app/services/lazykhschduler.py]: For lazyKh orginal code will be done after having a second working version

File: app/services/utils.py (hunk 8 above)
Comment (MohammedN-hash) [reply app/services/utils.py]: Renamed to utilities

File: app/services/utils.py (hunk 14 above)
Comment (MohammedN-hash) [reply app/services/utils.py]: For lazyKh orginal code will be done after having a second working version

File: app/services/utils.py (hunk 9 above)
Comment (MohammedN-hash) [reply app/services/utils.py]: For lazyKh orginal code will be done after having a second working version

**Review by MohammedN-hash — APPROVED** [review]

File: .env
Comment (wensle) [inline .env:1]: Delete this file. Don't commit.env files to the repo. Add **/.env (or similar) to .gitignore.

File: note
Comment (wensle) [reply note:1]: We use .env files now

File: app/models/pos.py [hunk 15]
```diff
@@ -0,0 +1,129 @@
+import spacy
+import re
+
+_nlp = spacy.load("en_core_web_sm")
```
Comment (wensle) [inline app/models/pos.py:4]: May need to download model first. Find out a way to do it via a Python function.

File: .env
Comment (wensle) [reply .env:1]: Fixed it in a push

**Comment by wensle** [comment]
@MohammedN-hash @shaaarplegs @Andreea2698 The Refactoring_code branch is now deleted, because of the merge. Make another branch as needed.
